data-raw/build-model-summary-refs.R

```diff
@@ -0,0 +1,45 @@
+library(dplyr)
+devtools::load_all() # can't call library(rbabylon) from within rbabylon when using pkgr
+
+###################
+# FUNCTION DEF
+###################
+
+#' Helper to re-build summary object and parameter table references
+#'
+#' Saves to `inst/test-refs/`.
+#' @param .mod bbi_nonmem_model object
+build_model_summary_refs <- function(.mod) {
+
+  .model_name <- get_model_id(.mod)
+  .ref_out_dir <- system.file('test-refs', package = 'rbabylon')
+  sum_out_path <- file.path(
+    .ref_out_dir,
+    as.character(glue::glue("{.model_name}_summary_obj.R"))
+  )
+  param_out_path <- file.path(
+    .ref_out_dir,
+    as.character(glue::glue("{.model_name}_param_table.R"))
+  )
+
+  withr::with_options(list(rbabylon.bbi_exe_path = read_bbi_path()), {
+    message(paste("  writing", sum_out_path))
+    .sum <- model_summary(.mod)
+    dput(.sum, file = sum_out_path)
+
+    message(paste("  writing", param_out_path))
+    param_df <- param_estimates(.sum)
+    dput(param_df, file = param_out_path)
+  })
+}
+
+###################
+# CALLS
+###################
+
+# To render refs, source this script.
+.proj_root <- rprojroot::find_rstudio_root_file()
+
+file.path(.proj_root, "inst/model/nonmem/basic", 1) %>%
```

Comment:
A major reason to use `file.path()` is to gain platform independence as a result of the `fsep` argument. Passing a string that contains platform-specific path separators, e.g., `inst/model/nonmem/basic`, negates this benefit. A call like

```r
file.path(.proj_root, "inst", "model", "nonmem", "basic", 1)
```

will preserve the platform independence.